File: DoAnXepLich/backend/XepLichGiangDay.Api/Models/Entities/NguoiDung.cs
namespace XepLichGiangDay.Api.Models.Entities;

public class NguoiDung
{
    public int MaNguoiDung { get; set; }
    public string TenDangNhap { get; set; } = string.Empty;
    public string MatKhau { get; set; } = string.Empty;
    public string HoTen { get; set; } = string.Empty;
    public string? Email { get; set; }
    public string VaiTro { get; set; } = "GiangVien"; // Admin, TruongBoMon, GiangVien
    public int? MaGiangVien { get; set; }
    public string TrangThai { get; set; } = "KichHoat"; // KichHoat, Khoa
}

File: DoAnXepLich/frontend/src/types/nguoiDung.ts
export interface NguoiDung {
  maNguoiDung: number
  tenDangNhap: string
  hoTen: string
  email?: string
  vaiTro: string // Admin, TruongBoMon, GiangVien
  maGiangVien?: number
  tenGiangVien?: string
  trangThai: string // KichHoat, Khoa
}

export interface TaoNguoiDungRequest {
  tenDangNhap: string
  matKhau: string
  hoTen: string
  email?: string
  vaiTro: string
  maGiangVien?: number
}

export interface CapNhatNguoiDungRequest {
  hoTen: string
  email?: string
  vaiTro: string
  maGiangVien?: number
  trangThai: string
}
